### PR TITLE
@storybook/addon-knobs: groupId

### PR DESCRIPTION
**Pull request by joscha**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/storybooks/storybook/issues/3374, 
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by martynaskadisa — APPROVED** [review]

File: types/storybook__addon-knobs/index.d.ts [hunk 1]
```diff
@@ -18,35 +18,37 @@ export interface StoryContext {
...
 }
 
-export interface NumberOptions {
+export type NumberOptions = {
     range: boolean;
     min: number;
     max: number;
     step: number;
-}
+} | {};
```
Comment (mhegazy) [inline types/storybook__addon-knobs/index.d.ts]: `{}` is a super type of any other type. so this type is basically `{}` which is almost `any`. 

did you mean to make the properties optional instead?

Comment (mhegazy) [reply types/storybook__addon-knobs/index.d.ts]: e.g.:

```ts
{
     range?: boolean;
     min?: number;
     max?: number;
     step?: number;
}
```

**Review by mhegazy — CHANGES_REQUESTED** [review]

File: types/storybook__addon-knobs/index.d.ts (hunk 1 above)
Comment (mhegazy) [reply types/storybook__addon-knobs/index.d.ts]: also switching from `interface` to `type` is a breaking change to users who can extend this type today. if you do not think users are doing this today, then it should be fine.

Comment (joscha) [reply types/storybook__addon-knobs/index.d.ts]: The `{}` is deliberate, we either need to pass all of them or the default options when omitting them (e.g. an empty object).
Regarding the interface change, I don't think people would extend, but you never know, so I will create an intermediate type.

Comment (mhegazy) [reply types/storybook__addon-knobs/index.d.ts]: > The {} is deliberate, we either need to pass all of them or the default options when omitting them (e.g. an empty object).

then write it as:
```ts
{
     range: boolean;
     min: number;
     max: number;
     step: number;
} | {
     range?: undefined;
     min?: undefined;
     max?: undefined;
     step?: undefined;
};

```

Comment (joscha) [reply types/storybook__addon-knobs/index.d.ts]: done, ptal

**Comment by mhegazy** [comment]
@joscha there are a couple of lint failures that were introduced. can you take a look.

**Comment by Fyre91** [comment]
Few know types like radio, button are missing definition, maybe it got introduced later in storybook.

**Pull request merged**